libs/audit-service/src/main/java/uk/gov/di/ipv/core/library/service/AuditService.java

```diff
@@ -2,48 +2,76 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
-import software.amazon.awssdk.services.sqs.SqsClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.amazon.awssdk.http.crt.AwsCrtAsyncHttpClient;
+import software.amazon.awssdk.services.sqs.SqsAsyncClient;
 import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse;
 import uk.gov.di.ipv.core.library.auditing.AuditEvent;
 import uk.gov.di.ipv.core.library.exceptions.SqsException;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
 import static software.amazon.awssdk.regions.Region.EU_WEST_2;
 import static uk.gov.di.ipv.core.library.config.EnvironmentVariable.SQS_AUDIT_EVENT_QUEUE_URL;
 
 public class AuditService {
-    private final SqsClient sqs;
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private final SqsAsyncClient sqs;
     private final String queueUrl;
     private final ObjectMapper objectMapper;
 
-    public AuditService(SqsClient sqs, ConfigService configService) {
+    private List<CompletableFuture<SendMessageResponse>> events = new ArrayList<>();
+
+    public AuditService(SqsAsyncClient sqs, ConfigService configService) {
         this.sqs = sqs;
         this.queueUrl = configService.getEnvironmentVariable(SQS_AUDIT_EVENT_QUEUE_URL);
         this.objectMapper = new ObjectMapper();
     }
 
-    public AuditService(SqsClient sqs, ConfigService configService, ObjectMapper objectMapper) {
+    public AuditService(
+            SqsAsyncClient sqs, ConfigService configService, ObjectMapper objectMapper) {
         this.sqs = sqs;
         this.queueUrl = configService.getEnvironmentVariable(SQS_AUDIT_EVENT_QUEUE_URL);
         this.objectMapper = objectMapper;
     }
 
-    public static SqsClient getSqsClient() {
-        return SqsClient.builder()
+    public static SqsAsyncClient getSqsClient() {
+        return SqsAsyncClient.builder()
                 .region(EU_WEST_2)
-                .httpClientBuilder(UrlConnectionHttpClient.builder())
+                .httpClientBuilder(AwsCrtAsyncHttpClient.builder())
```

Comment:
The AWS docs seem to imply that we should use the AWS CRT client for the sync case as well (https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/http-configuration.html#http-clients-recommend) - unless it's a snapstart thing to prefer the `UrlConnectionHttpClient`?